services/src/main/java/org/keycloak/protocol/saml/IDPMetadataDescriptor.java

```diff
@@ -76,9 +78,13 @@ public static String getIDPDescriptor(URI loginPostEndpoint, URI loginRedirectEn
 
         spIDPDescriptor.addSingleLogoutService(new EndpointType(SAML_HTTP_POST_BINDING.getUri(), logoutEndpoint));
         spIDPDescriptor.addSingleLogoutService(new EndpointType(SAML_HTTP_REDIRECT_BINDING.getUri(), logoutEndpoint));
+        spIDPDescriptor.addSingleLogoutService(new EndpointType(SAML_HTTP_ARTIFACT_BINDING.getUri(), logoutEndpoint));
         spIDPDescriptor.addSingleSignOnService(new EndpointType(SAML_HTTP_POST_BINDING.getUri(), loginPostEndpoint));
         spIDPDescriptor.addSingleSignOnService(new EndpointType(SAML_HTTP_REDIRECT_BINDING.getUri(), loginRedirectEndpoint));
         spIDPDescriptor.addSingleSignOnService(new EndpointType(SAML_SOAP_BINDING.getUri(), loginPostEndpoint));
+        spIDPDescriptor.addSingleSignOnService(new EndpointType(SAML_HTTP_ARTIFACT_BINDING.getUri(), loginPostEndpoint));
+
+        spIDPDescriptor.addArtifactResolutionService(new IndexedEndpointType(SAML_SOAP_BINDING.getUri(), artifactResolutionService));
```

Comment:
After your commit which Move artifact resolution into separate endpoint, this won't be correct

Reply:
It should be handled here: https://github.com/keycloak/keycloak/pull/7756/commits/a7d4d8e27f800844242eba9595e4d70977d058f8#diff-07d4e837666bc51baf1a786aaecee93ea834dfee1895f9fc826a1109ffc61784R859. Currently that line should append "/resolve" to the end of saml protocol url.

Reply:
@mhajas I had a look to at the changes you added. The way you have implemented the different logout binding than login for artifact binding is ok for me. It perfectly solves my comment, thanks !

Reply:
Sorry, I misread it.

Reply:
np, thank you for checking.